src/core_plugins/kibana/public/home/components/recently_accessed.js

```diff
@@ -0,0 +1,188 @@
+      <EuiLink onClick={this.onButtonClick.bind(this)}>
+        <EuiTextColor color="subdued">
+          {`+${dropdownLinks.length}`}
+        </EuiTextColor>
+        <EuiIcon
+          type="arrowDown"
+          color="subdued"
+        />
+      </EuiLink>
+    );
+
+    return (
+      <EuiFlexItem
+        key="dropdown"
+        grow={false}
+      >
+        <EuiPopover
+          id="popover"
+          ownFocus
+          button={openPopoverComponent}
+          isOpen={this.state.isPopoverOpen}
+          closePopover={this.closePopover.bind(this)}
+          anchorPosition="downRight"
+        >
+          <ul>
+            {dropdownLinks}
+          </ul>
+        </EuiPopover>
+      </EuiFlexItem>
+    );
+  }
+
+  renderLongLink = (link, label, includeSeperator = false) => {
+    let seperator;
+    if (includeSeperator) {
+      seperator = (
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type="dot"
+            color="subdued"
+          />
+        </EuiFlexItem>
+      );
+    }
+    // Want to avoid a bunch of white space around items with short labels (happens when min width is too large).
+    // Also want to avoid truncating really short names (happens when there is no min width)
+    // Dynamically setting the min width based on label lengh meets both of these goals.
+    const EM_RATIO = 0.6; // 'em' ratio that avoids too much horizontal white space and too much truncation
+    const minWidth = (label.length < 8 ? label.length : 8) * EM_RATIO;
+    const style = {
+      overflow: 'hidden',
+      minWidth: `${minWidth}em`
+    };
+    return (
+      <React.Fragment key={link}>
+        {seperator}
+        <EuiFlexItem
+          style={style}
+          grow={false}
+        >
+          <EuiLink
+            className="longLink"
+            href={link}
+          >
+            {label}
+          </EuiLink>
+        </EuiFlexItem>
+      </React.Fragment>
+    );
+  }
+
+  renderRecentlyAccessed = () => {
+    if (this.props.recentlyAccessed.length <= NUM_LONG_LINKS) {
+      return this.props.recentlyAccessed.map((item, index) => {
+        let includeSeperator = true;
+        if (index === 0) {
+          includeSeperator = false;
+        }
+        return this.renderLongLink(item.link, item.label, includeSeperator);
+      });
+    }
+
+    const links = [];
+    for (let i = 0; i < NUM_LONG_LINKS; i++) {
+      let includeSeperator = true;
+      if (i === 0) {
+        includeSeperator = false;
+      }
+      links.push(this.renderLongLink(
+        this.props.recentlyAccessed[i].link,
+        this.props.recentlyAccessed[i].label,
+        includeSeperator));
+    }
+
+    return [
+      ...links,
+      this.renderDropdown()
+    ];
+  };
+
+  render() {
+    return (
+      <EuiPanel paddingSize="l">
+        <EuiText>
+          <p>
+            <EuiTextColor color="subdued">
+              Recently accessed
+            </EuiTextColor>
+          </p>
+        </EuiText>
```

Comment:
This looks a little tight. How about a small spacer beneath this title?

![image](https://user-images.githubusercontent.com/1238659/36004969-29f24a22-0cea-11e8-8004-a655511ee8ca.png)

Reply:
good idea. Spacer has been added.